product/host.go

```diff
@@ -22,22 +26,37 @@ func NewHost(logger hclog.Logger, cfg Config, hcl2 *hcl.Host) (*Product, error)
 	if cfg.OS == "auto" {
 		os = runtime.GOOS
 	}
-	// TODO(mkcp): Host can have an API client now and it would simplify quite a bit.
-	product.Runners = hostRunners(os)
+
 	if hcl2 != nil {
-		hclRunners, err := hcl.BuildRunners(hcl2, cfg.TmpDir, nil, cfg.Since, cfg.Until)
+		// Map product-specific redactions from our config
+		hclProductRedactions, err := hcl.MapRedacts(hcl2.Redactions)
+		if err != nil {
+			product.l.Error("problem mapping Host redactions from HCL config")
+		}
+
+		// Prepend product HCL redactions to our product defaults
+		cfg.Redactions = append(hclProductRedactions, cfg.Redactions...)
+
+		hclRunners, err := hcl.BuildRunners(hcl2, cfg.TmpDir, nil, cfg.Since, cfg.Until, cfg.Redactions)
 		if err != nil {
 			return nil, err
 		}
 		product.Runners = append(product.Runners, hclRunners...)
 		product.Excludes = hcl2.Excludes
 		product.Selects = hcl2.Selects
 	}
+
+	// TODO(mkcp): Host can have an API client now and it would simplify quite a bit.
+	// Add built-in runners
+	builtInRunners := hostRunners(os, cfg.Redactions)
+	product.Runners = append(product.Runners, builtInRunners...)
+
 	return product, nil
 }
 
 // hostRunners generates a slice of runners to inspect the host.
-func hostRunners(os string) []runner.Runner {
+func hostRunners(os string, redactions []*redact.Redact) []runner.Runner {
+	// TODO(dcohen) Looks like none of these runners do redactions yet
```

Comment:
Yeah - host runners are so unlikely to reveal customer data that threading the redactions through are the last step. If any of these are higher order runners that implement commander or sheller then their impl. will already be taken care of, and we just need to pass in the redact data.

Reply:
Thanks, makes sense!